lib/vmselectapi/server.go

```diff
@@ -931,8 +937,13 @@ func (s *Server) processSeriesSearch(ctx *vmselectRequestCtx) error {
 	if err := bi.Error(); err != nil {
 		return fmt.Errorf("search error: %w", err)
 	}
-	ctx.qt.Printf("sent %d blocks to vmselect", blocksRead)
 
+	s.seriesPerQueryRead.Update(float64(len(seriesStats)))
```

Comment:
This histogram shows the number of unique series found per each `vmstorage`. While this metric can be useful for estimating per-query memory usage at `vmstorage`, I doubt it will be useful for estimating per-query memory usage at `vmselect`. It is better creating the same histogram at `vmselect` after it collects data from all the `vmstorage` nodes.